packages/ui/package.json

```diff
@@ -4,12 +4,12 @@
   "license": "MIT",
   "type": "module",
   "scripts": {
-    "assets:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src/",
+    "prebuild:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src/ && cp index.d.ts ./dist-native/src/",
```

Comment:
@fbwoolf wonder if we can move this asset moving from package.json to the build tool itself, be in metro or tsup

Reply:
It wasn't working for me using either until I added this. There seems to be issues (listed in description) open with tsup and loading svgs. The svgr plugin hits errors but I can try again for mobile. We aren't currently using tsup to build for native. Right now, I'm running into issue using tsup for the web build. I'm also unclear as to why we have a `dist-all` and a `dist-web`?

We are using the metro config in the UI lib, but I'm unclear how/if it is used?

Reply:
@edgarkhanzadian set this up. I'm not sure if we need `dist-all` or if this is required for publication purposes.

Either way, wouldn't concern yourself with current set up. Delete it and start again if this helps you solve problems.

Reply:
@edgarkhanzadian any reason we need to use metro for native bundling? Couldn't we use tsup for this as well, and metro in mobile will handle RN specific bundling concerns?

Reply:
`dist-all` is being used as the package export for web atm. I'm actually not sure if `dist-web` is being used or can just replace `dist-all`? I'm currently working through changes for web to get the svg export working for the extension, so @edgarkhanzadian any feedback you can provide will be helpful. I'm making changes but don't want to overlook issues you've already confronted.

Reply:
@kyranjamie it appears there is a slightly problematic lib being used in tsup that is used to gen types when `dts` is true:
https://github.com/egoist/tsup/issues/993
https://github.com/Swatinem/rollup-plugin-dts/issues/277

In docs, there is an experimental option I am going to try using:
https://tsup.egoist.dev/#generate-declaration-file